bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -105,6 +121,20 @@ class SourceSelector {
 
   uint16_t minPrePrepareMsgsForPrimaryAwarness() { return minPrePrepareMsgsForPrimaryAwarness_; }
 
+  // Metric
+  inline void updatePreferredReplicasMetric() const {
+    metrics_.preferred_replicas_.Get().Set(preferredReplicasToString());
+  }
+  inline void updatePreferredReplicasMetric(std::string replicas) const {
+    metrics_.preferred_replicas_.Get().Set(replicas);
+  }
+  inline void updateCurrentReplicaMetric() const { metrics_.current_source_replica_.Get().Set(currentReplica_); }
```

Comment:
The above 3 can be removed in my opinion.
Call all explicitly in SS code whenever you change preffered replicas / current source.
It adds nothing to the code, actually it reduce readibilty since its clearer to see in the code what you are doing - there is no logic here, and you don't dave anything on code-reuse,  it's a 1 liner. Also, they were all added as a public interface while we shouldn't update the metrics from outside.